src/teachy_tv.c

```diff
@@ -0,0 +1,1781 @@
+        ");
+}
+#endif
+
+void TeachyTvClusFuncTransitionRenderBg2TeachyTvGraphicInitNpcPos(u8 taskId)
+{
+    u16 *data;
+    u32 counter;
+
+    data = (u16 *)gTasks[taskId].data;
+    TeachyTvBg2AnimController();
+    counter = data[2] + 1;
+    data[2] = counter;
+    if ( (s16)counter > 63 )
+    {
+        CopyToBgTilemapBufferRect_ChangePalette(2u, (u8 *)gUnknown_203F450 + 0x3004, 0, 0, 0x20u, 0x20u, 0x11u);
+        TeachyTvSetSpriteCoordsAndSwitchFrame(((u8*)data)[2], 8, 0x38, 7u);
+        schedule_bg_copy_tilemap_to_vram(2u);
+        data[2] = 0;
+        ++data[3];
+        PlayNewMapMusic(BGM_FRLG_FOLLOW_ME);
+    }
+}
+
+void TeachyTvClusFuncClearBg2TeachyTvGraphic(u8 taskId)
+{
+    u16 *data;
+    u32 counter;
+
+    data = (u16*)gTasks[taskId].data;
+    counter = data[2] + 1;
+    data[2] = counter;
+    if ( (s16)counter == 134 )
+    {
+        FillBgTilemapBufferRect_Palette0(2u, 0, 2u, 1u, 0x1Au, 0xCu);
+        schedule_bg_copy_tilemap_to_vram(2u);
+        data[2] = 0;
+        ++data[3];
+    }
+}
+
+void TeachyTvClusFuncNpcMoveAndSetupTextPrinter(u8 taskId)
+{
+    s16 *data;
+    struct Sprite * spriteAddr;
+
+    data = gTasks[taskId].data;
+    spriteAddr = &gSprites[data[1]];
+    if ( data[2] != 35 )
+        ++data[2];
+    else {
+        if ( spriteAddr->pos2.x == 0x78 )
+        {
+            StartSpriteAnim(&gSprites[data[1]], 0);
+            TeachyTvInitTextPrinter(&gUnknown_841B83D);
+            data[2] = 0;
+            ++data[3];
+        }
+        else
+            ++spriteAddr->pos2.x;
+    }   
+}
+
+void TeachyTvClusFuncIdleIfTextPrinterIsActive(u8 taskId)
+{
+    u16* data = (u16 *)gTasks[taskId].data;
+    if ( !(sub_80BF518(0) << 16) )
+        ++data[3];
+}
+
+void TeachyTvRenderMsgAndSwitchClusterFuncs(u8 taskId)
+{
+    u16 *data;
+    data = gTasks[taskId].data;
+    if ( gMain.newKeys & 2 )
+    {
+        char *offset = ((char *)gUnknown_203F450 + 0x4006);
+        u32 v4 = 0;
+        offset[0] = 1;
+        TeachyTvSetSpriteCoordsAndSwitchFrame(*((char *)data + 2), 0, 0, 0);
+        FillWindowPixelBuffer(0, 0xCCu);
+        CopyWindowToVram(0, 2u);
+        TeachyTvClearBg1EndGraphicText();
+        data[2] = v4;
+        data[3] = v4;
+        gTasks[taskId].func = TeachyTvBackToOptionList;
+    }
+    else
+    {
+        void (***array)(u8) = &gUnknown_8479548;
+        void (**cluster)(u8) = array[(u8)gTeachyTV_StaticResources.optionChosen];
+        _call_via_r1(
+            taskId,
+            cluster[(s16)data[3]]);
+    }
+}
+
+void TeachyTvClusFuncTextPrinterSwitchStringByOptionChosen(u8 taskId)
+{
+    u16 *data = (u16 *)gTasks[taskId].data;
+    char ** texts = &gUnknown_8479560;
+    TeachyTvInitTextPrinter(texts[gTeachyTV_StaticResources.optionChosen]);
+    ++data[3];
+}
+
+void TeachyTvClusFuncTextPrinterSwitchStringByOptionChosen2(u8 taskId)
+{
+    u16 *data = (u16 *)gTasks[taskId].data;
+    char ** texts = &gUnknown_8479578;
```

Comment:
Is this declaration actually necessary?  If so, should be `const char *const *texts`